From be2edd2b68f01af344c6e6c02ba0ffa1b6f1fbfd Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Sun, 8 Jan 2017 17:05:05 +0200
Subject: [PATCH 0714/1345] fix: dts: cp110: fix interrupt mapping for eip197

commit  11d13937fb037c822c2febdafc0f0dff9e385955 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

EIP197 engine was tested only on CP0.
The available interrupts mapping for CP1 is not correct.
This patch fixes the interrupt mapping, this allows to use EIP197
on CP1 too.

In addition, add eip197 to cp110-1 (disabled) in configuration
'A' and 'D'.
It's disabled because, currently, EIP197 can't run on both CPs
in the same time, so we add it as a place holder.

Change-Id: I40e25ee9e9e8da31cc2067696bf7a6e5a554a986
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35377
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db-A.dts |    3 +++
 arch/arm64/boot/dts/marvell/armada-8040-db-D.dts |    3 +++
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi  |    7 ++++++-
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
index 1f54584..017e97b 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-A.dts
@@ -211,6 +211,9 @@
 					status = "disabled";
 				};
 			};
+			eip197@800000 {
+				status = "disabled";
+			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
index f7c4cb6..67c5fe7 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-D.dts
@@ -203,6 +203,9 @@
 					status = "disabled";
 				};
 			};
+			eip197@800000 {
+				status = "disabled";
+			};
 			comphy {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index ae3c925..3eff70e 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -355,7 +355,12 @@ eip197@800000 {
 	compatible = "marvell,eip197";
 	reg = <0x800000 0x200000>;
 	dma-coherent;
-	interrupts = <ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>;
+	interrupts = <ICU_GRP_NSR 88 IRQ_TYPE_LEVEL_HIGH>,
+			<ICU_GRP_NSR 89 IRQ_TYPE_LEVEL_HIGH>,
+			<ICU_GRP_NSR 90 IRQ_TYPE_LEVEL_HIGH>,
+			<ICU_GRP_NSR 91 IRQ_TYPE_LEVEL_HIGH>,
+			<ICU_GRP_NSR 92 IRQ_TYPE_LEVEL_HIGH>;
+	interrupt-names = "ring0", "ring1", "ring2", "ring3", "global";
 	clocks = <&cps_syscon0 1 26>;
 	status = "disabled";
 };
-- 
1.7.9.5

